src/clang-cpp-frontend/clang_cpp_convert.h

```diff
@@ -16,14 +16,46 @@ class clang_cpp_convertert : public clang_c_convertert
   virtual ~clang_cpp_convertert() = default;
 
 protected:
+  typedef std::unordered_map<std::size_t, std::pair<std::string, typet>>
+    this_mapt;
+  this_mapt this_map;
+
   bool get_decl(const clang::Decl &decl, exprt &new_expr) override;
 
+  // Get reference to declared variabales or functions, e.g:
+  //  - getting the callee for CXXConstructExpr
+  //  - getting the object for DeclRefExpr
+  bool get_decl_ref(const clang::Decl &decl, exprt &new_expr);
+
   bool get_type(const clang::QualType &type, typet &new_type) override;
 
   bool get_type(const clang::Type &the_type, typet &new_type) override;
 
   bool get_function(const clang::FunctionDecl &fd, exprt &new_expr) override;
 
+  // Get reference for constructor callsite
+  bool get_constructor_call(
+    const clang::CXXConstructExpr &constructor_call,
+    exprt &new_expr);
+
+  bool
+  get_function_body(const clang::FunctionDecl &fd, exprt &new_expr) override;
+
+  // Get function params for C++
+  // contains parsing routines specific to C++ class member functions
+  bool get_function_params(
+    const clang::FunctionDecl &fd,
+    code_typet::argumentst &params) override;
+
+  // Add implicit `this' when parsing C++ class member functions, e.g:
```

Comment:
Can I ask you to follow the standard for comments as in: https://github.com/esbmc/esbmc/blob/master/src/goto-symex/goto_symex_state.h?